src/it/projects/MJAVADOC-498_modulepath/verify.groovy

```diff
@@ -17,5 +17,6 @@
  * under the License.
  */
 def log = new File( basedir, 'build.log').text
-
-assert log.count( " --module-path" ) == 3
+def count = log.count( " --module-path" )
+// depends on Maven version
+assert count == 2 || count == 3
```

Comment:
We have a `mavenVersion` property defined so we can add something like 
```
assert mavenVersion.startWith('4') && count == x || mavenVersion.startWith('3') && count == y
```

Reply:
OK, let me try that. But how to solve before 3.9.x and after? That is a mere string.

Reply:
yes it is a string

Reply:
I improved the comment and will leave as-is. Without a selector on an object that is too much fiddling.